src/index.ts

```diff
@@ -1,22 +1,28 @@
-import assert from 'assert';
-import ExtendableError from 'es6-error';
-
+import * as assert from 'assert';
+import ExtendableError from 'extendable-error';
 const isString = d => Object.prototype.toString.call(d) === '[object String]';
 const isObject = d => Object.prototype.toString.call(d) === '[object Object]';
-
+interface ErrorConfig {
+  message: string;
+  time_thrown: any;
+  data: any,
+  options: any,
+}
 class ApolloError extends ExtendableError {
-  constructor (name, {
-    message,
-    time_thrown = (new Date()).toISOString(),
-    data = {},
-    options = {},
-  }) {
-    const t = (arguments[2] && arguments[2].time_thrown) || time_thrown;
-    const d = Object.assign({}, data, ((arguments[2] && arguments[2].data) || {}));
-    const m = (arguments[2] && arguments[2].message) || message;
-    const opts = Object.assign({}, options, ((arguments[2] && arguments[2].options) || {}));
-
-    super(m);
+  name: string;
+  message: string;
+  time_thrown: any;
+  data: any;
+  path: any;
+  locations: any;
+  _showLocations: any;
```

Comment:
We should probably type this as `boolean` and make sure that it is set or defaulted to false